src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/extension.ts

```diff
@@ -164,6 +170,9 @@ export async function activate(vscodeType: typeof vscodeapi, context: ExtensionC
                 vscodeType.languages.registerRenameProvider(
                     RazorLanguage.id,
                     renameProvider),
+                vscodeType.languages.registerDocumentHighlightProvider(
```

Comment:
Instead of registering this here as a secondary provider you could start down the single-server approach and allow the Razor side to fill out highlights like it does in VS. You'd need to add the TypeScript to delegate the highlight calls but it'd work!

Reply:
Just tried out the single server approach with both hover and document highlight. The former worked, but the latter still has issues with the `executeCommand` bit which makes me think that VSCode's document highlight command is simply buggy somehow ☹️